### PR TITLE
Workplace endpoins

### LINKED ISSUE #33: Workplace endpoint

### PR DESCRIPTION
**Pull request by Bazilis**

### CODE REVIEW

File: src/Api/Controllers/WorkplaceController.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        private readonly IWorkplaceService _workplaceService;
+        public WorkplaceController(IWorkplaceService workplaceService)
+        {
+            _workplaceService = workplaceService;
+        }
+
+        // GET: api/<WorkplaceController>
+        [HttpGet]
+        [Produces("application/json")]
+        public async Task<ActionResult<IEnumerable<WorkplaceDto>>> GetAll([FromQuery] WorkplaceFilterDto filterModel)
```
Comment (khamza85) [inline src/Api/Controllers/WorkplaceController.cs:22]: You can simplify return type to a shorter alternative i.e. `Task<WorkplaceDto[]>`

**Review by SandroShera — APPROVED** [review]
Very helpful

File: src/Api/Controllers/WorkplaceController.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+            if (workplaceUpdated == null)
+                return BadRequest("Input model is null");
+
+            return Ok(workplaceUpdated);
+        }
+
+        // DELETE api/<WorkplaceController>/5
+        [HttpDelete("{id}")]
+        public async Task<ActionResult> Delete(Guid id)
```
Comment (khamza85) [inline src/Api/Controllers/WorkplaceController.cs:74]: `Task<IActionResult>`

File: src/Api/Controllers/WorkplaceController.cs (hunk 1 above)
Comment (Bazilis) [reply src/Api/Controllers/WorkplaceController.cs]: Fixed

File: src/Api/Services/WorkplaceService.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        private readonly AppDbContext _context;
+
+        public WorkplaceService(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<WorkplaceDto>> GetWorkplaces(WorkplaceFilterDto filterModel)
+        {
+            var workplaces = _context.Workplaces.AsQueryable();
```
[the remark this answers is not in the document]
Comment (Bazilis) [reply src/Api/Services/WorkplaceService.cs]: Fixed

File: src/Api/Controllers/WorkplaceController.cs (hunk 2 above)
Comment (Bazilis) [reply src/Api/Controllers/WorkplaceController.cs]: Fixed

**Review by apfpo — APPROVED** [review]

**Pull request merged**